INSTALL

```diff
@@ -11,12 +24,31 @@ java -jar apache-rat.jar -E ./.rat-excludes -d .
 # to connect to other services. You might want to test or run Airflow
 # from a virtual env to make sure those dependencies are separated
 # from your system wide versions
-python -m my_env
-source my_env/bin/activate
 
-# [required] building and installing
-# by pip (preferred)
+python3 -m venv PATH_TO_YOUR_VENV
+source PATH_TO_YOUR_VENV/bin/activate
+
+# [required] building and installing by pip (preferred)
 pip install .
 
 # or directly
 python setup.py install
+
+# You can also install recommended version of the dependencies by using requirements.txt
+# as constraint file
+
+pip install . --constraint requirements.txt
```

Comment:
How should I know if I should use requirements or not?

Reply:
You do not have to - it's the recommended version of dependencies. This is just a hint that you can do it. I proposed to have "apache-airflow-pinned" package release separately. with clear purpose in mind "to have repeatable installation"

I agreed  having the --constraint and having to know which version to install might be confusing - that was my point in the discussion - but there are different opinions. For now I just want to merge requirements.txt and it's maintenance mechanism so that we avoid further breakage. But the discussion continues here how to approach it:

https://lists.apache.org/thread.html/rf8b0ea9a7b93482734ae9e10e7d87f3c31679edb5eaf88496e733aa5%40%3Cdev.airflow.apache.org%3E

I invite you to take part in the discussion and propose some solutions/approach. I would really love to have yours (and others) opinions in the thread there.